src/lib/ast.rs

```diff
@@ -695,14 +695,15 @@ pub fn parse_file(input_path: &str) -> Result<Arena<String>, ParseError> {
         Some(ext) => ext.to_str().unwrap(),
         None => return Err(ParseError::NoFileExtension),
     };
-    let lex_l_path = format!("grammars/{}.l", extension);
-    let yacc_y_path = format!("grammars/{}.y", extension);
+    let mut lex_l_path = format!("grammars/{}.l", extension);
+    let mut yacc_y_path = format!("grammars/{}.y", extension);
     info!("Using lexer: {} and parser: {} for input: {}",
           &lex_l_path,
           &yacc_y_path,
           input_path);
     if !Path::new(&lex_l_path).exists() || !Path::new(&yacc_y_path).exists() {
-        return Err(ParseError::FileNotFound);
+        lex_l_path = String::from("grammars/txt.l");
```

Comment:
I suspect that `std::fs::canonicalize` will be part (but not all) of the solution.